tck/methods/key.ts

```diff
@@ -0,0 +1,207 @@
+import { KeyGenerationResponse } from "../response/key";
+
+const generateKey = (params: KeyGenerationParams): KeyGenerationResponse => {
+    if (
+        params.fromKey &&
+        ![
+            AccountKey.ED25519_PUBLIC_KEY,
+            AccountKey.ECDSA_SECP256K1_PUBLIC_KEY,
+            AccountKey.EVM_ADDRESS,
+        ].includes(params.type as AccountKey)
+    ) {
+        return invalidParamError<KeyGenerationResponse>(
+            "fromKey should only be provided for specific public key types or EVM address",
+        );
+    }
+
+    if (params.threshold && params.type !== AccountKey.THRESHOLD_KEY) {
+        return invalidParamError<KeyGenerationResponse>(
+            "threshold should only be provided for thresholdKey types",
+        );
+    }
+
+    if (
+        params.keys &&
+        ![AccountKey.KEY_LIST, AccountKey.THRESHOLD_KEY].includes(
+            params.type as AccountKey,
+        )
+    ) {
+        return invalidParamError<KeyGenerationResponse>(
+            "keys should only be provided for keyList or thresholdKey types",
+        );
+    }
+
+    if (
+        (params.type === AccountKey.THRESHOLD_KEY ||
+            params.type === AccountKey.KEY_LIST) &&
+        !params.keys
+    ) {
+        return invalidParamError<KeyGenerationResponse>(
+            "keys list is required for generating a KeyList type",
+        );
+    }
+
+    if (params.type === AccountKey.THRESHOLD_KEY && params.threshold === 0) {
+        return invalidParamError<KeyGenerationResponse>(
+            "threshold is required for generating a ThresholdKey type",
+        );
+    }
+
+    const response: KeyGenerationResponse = {
+        key: "",
+        privateKeys: [],
+    };
+
+    try {
+        const { key, privateKeys } = processKeyRecursively(
+            params,
+            response,
+            false,
+        );
+
+        response.key = key;
+        response.privateKeys = privateKeys;
+    } catch (error) {
+        return invalidParamError<KeyGenerationResponse>(error.message);
+    }
+
+    return response;
+};
+
+const processKeyRecursively = (
+    params: KeyGenerationParams,
+    response: KeyGenerationResponse,
+    isList: boolean,
+): KeyGenerationResponse => {
+    switch (params.type) {
+        case AccountKey.ED25519_PRIVATE_KEY:
+            const ed25519PrivateKey = PrivateKey.generateED25519();
+            if (isList)
+                response.privateKeys.push(ed25519PrivateKey.toStringDer());
+            return {
+                key: ed25519PrivateKey.toStringDer(),
+                privateKeys: response.privateKeys,
+            };
+
+        case AccountKey.ECDSA_SECP256K1_PRIVATE_KEY:
+            const ecdsaPrivateKey = PrivateKey.generateECDSA();
+            if (isList)
+                response.privateKeys.push(ecdsaPrivateKey.toStringDer());
+            return {
+                key: ecdsaPrivateKey.toStringDer(),
+                privateKeys: response.privateKeys,
+            };
+
```

Comment:
Actually impressed that you used recursion. 200 IQ move here but can we use something simpler? 

Chances are at some points its going to lead to the TCK, eating someone's Mac memory and then he is going to create an issue that we have to resolve.

Reply:
If you think this is the cleanest approach, I am eager to accept it tho.

Reply:
The json rpc request has a recursive structure and it's easier to do it with recursion.